### PR TITLE
Updates to add zero_field utility to the track matching

### PR DESCRIPTION
**Pull request by david-stewart**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

File: offline/packages/trackbase/TrackFitUtils.cc [hunk 1]
```diff
@@ -862,3 +870,68 @@ float TrackFitUtils::get_helix_surface_pathlength(const Surface& surf, std::vect
...
+// phi, eta, pT, pos_dca, momentum -- note that momentum is such that pT\equiv1
+std::tuple<bool, double, double, double, Acts::Vector3, Acts::Vector3> 
+TrackFitUtils::zero_field_track_params(
+    ActsGeometry* _tGeometry, 
+    TrkrClusterContainer* _cluster_map, 
+    const std::vector<TrkrDefs::cluskey>& cluskey_vec)
+{
+  std::vector<Acts::Vector3> global_vec;
+
+  if (global_vec.size()<2)
```
Comment (hupereir) [inline offline/packages/trackbase/TrackFitUtils.cc:891]: Dave, 
Nice commit !
I'm confused though: global_vec is declared on line 889, empty. So the test here is always satisfied. No ? What do I miss ?

Comment (david-stewart) [reply offline/packages/trackbase/TrackFitUtils.cc:891]: I just saw this! It is an error, and I have moved it to after the call that is supposed to fix it.

File: offline/packages/trackbase/TrackFitUtils.cc [hunk 2]
```diff
@@ -862,3 +870,68 @@ float TrackFitUtils::get_helix_surface_pathlength(const Surface& surf, std::vect
...
+  Acts::Vector3 cluster = global_vec.back();
+  auto clus_dca_xy = get_line_point_pca(xy_m, xy_b, {cluster.x(),cluster.y(),0});
+  double px = clus_dca_xy.x();
+  double py = clus_dca_xy.y();
+  double pz = xz_m * x + xz_b;
+
+  // correct to direction from the pca to origin
+  px -= x;
+  py -= y;
+  pz -= z;
```
Comment (xuanli23) [inline offline/packages/trackbase/TrackFitUtils.cc:926]: For this commit, the pz will be always 0, is this in purpose or a typo?

**Comment by osbornjd** [comment]
Marked as draft while other comments are being resolved

**Review by osbornjd — COMMENTED** [review]
One other question/comment - this PR indicates we will not be actually introducing an object to store zero field track parameters and rather be recalculating them on the fly with the function in `TrackFitUtils` whenever they're needed. I don't have an issue with this because the zero field running is a niche "calibration" case which doesn't have stringent CPU requirements, but I wanted to make sure that was the plan moving forward as we should then replace the evaluation zero field machinery with this call to properly calculate the straight line fit track parameters

**Comment by sPHENIX-GitHub-l** [comment]
Hi Joe,

Yes, that is the plan. The zero field running requires changes in only two places (track matching and vertex finding). It did not seem to justify complicating the seed storage, given that we do not need to process much zero field data.

Tony


________________________________

File: offline/packages/trackbase/TrackFitUtils.cc (hunk 2 above)
Comment (david-stewart) [reply offline/packages/trackbase/TrackFitUtils.cc:926]: This is a typo. Line 921 should use `px` instead of `x`. Thanks, this is a good catch

**Pull request merged**